connectors/byoc.py

```diff
@@ -700,6 +700,10 @@ async def sync(self, sync_job_index, source_klass, elastic_server, bulk_options)
             await data_provider.ping()
             await asyncio.sleep(0)
 
+            logger.debug(f"Validating configuration for {data_provider}")
+            await data_provider.validate_configuration()
```

Comment:
The patch looks good,. I just have one doubt. Should we validate the configurations before this [line](https://github.com/elastic/connectors-python/pull/552/files#diff-b9a5d063adca223525457d0a3c06df668bd814fe46069dda6678b694c277edc9R700), i.e. before the ping ? 
The source class should be sure of the validity of the configurations before invoking the ping IMO.

Reply:
Good call, swapped in https://github.com/elastic/connectors-python/pull/552/commits/3eb88f4c21ff9e59fecd63040316706018e64449